core/src/main/java/com/linecorp/armeria/server/ServerPort.java

```diff
@@ -41,9 +42,28 @@
  */
 public final class ServerPort implements Comparable<ServerPort> {
 
+    private static final AtomicLong nextPortGroup = new AtomicLong();
+
+    /**
+     * Returns a unique value that is used for identifying a group of {@link ServerPort}s.
+     * When two ephemeral {@link ServerPort}s have the same port group value, {@link Server}
+     * will choose the same port number for them, rather than allocating two ephemeral ports.
+     */
+    static long nextPortGroup() {
+        for (;;) {
+            final long portGroup = nextPortGroup.incrementAndGet();
```

Comment:
Do we really need this for loop?
Can't we just return `nextPortGroup.incrementAndGet();`?

Reply:
Theoretically, another thread will generate group ID Long.MAX_VALUE times.

Reply:
Because we even use `AtomicLong` (not integer), I thought there are almost 0 chances that it overflows. 😄 
But yeah, I'm fine with it.